payloads/libpayload/arch/x86/head.S

```diff
@@ -44,7 +44,7 @@ _entry:
 	.align 4
 
 #define MB_MAGIC 0x1BADB002
-#define MB_FLAGS 0x00010003
+#define MB_FLAGS 0x00000003
```

Comment:
offset 12-28 have been initialized at line 53 to line 57, they are all filled with symbols address. This multiboot header has already provided info about header/load/load end/bass end and entry address, according to multiboot spec, loader can parse these fields and take corresponding action.
If loader does not want to use these info, loader can jump to the start address of memory where payload is copied, since instruction at the start of  binary is "jump to real entry", the precondition is that payload should support PIC/PIE.

Reply:
does this work?
It doesn't work under grub.

Reply:
payload in coreboot woks in baremetal and qemu. also this changes breaks multiboot header definition:
magic + flag + checksum = 0.
let us discussed this patch next week.

Reply:
line 53 to line 57 only defines:
12	u32	header_addr	if flags[16] is set
16	u32	load_addr	if flags[16] is set
20	u32	load_end_addr	if flags[16] is set
24	u32	bss_end_addr	if flags[16] is set

entry address is not specified

Reply:
checksum is calculated by
.long -(MB_MAGIC + MB_FLAGS)
 automatically

Reply:
Based on discussion this afternoon, I think root cause of zeroed header address should be caused by: 
CONFIG_LP_BASE_ADDRESS is unset.
According to link script payloads/libpayload/arch/x86/libpayload.ldscript, please have a try on set CONFIG_LP_BASE_ADDRESS  with specified address.

Reply:
you are right. I tried it works.

Let's hold on this patch.